### PR TITLE
Run analyzers initial impl

### PR DESCRIPTION
**Pull request by jmarolf**
making anaysis happen in parallel

### CODE REVIEW

File: src/Analyzers/CodeAnalysisResult.cs [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+                     return list;
+                 },
+                updateValueFactory: (key, list) =>
+                {
+                    list.Add(diagnostic);
+                    return list;
+                });
+        }
+
+        public IReadOnlyDictionary<DocumentId, ImmutableArray<Diagnostic>> Diagnostics
```
Comment (RikkiGibson) [inline src/Analyzers/CodeAnalysisResult.cs]: I need this to proceed with PerDocumentCodeFixApplier

File: src/Analyzers/ConcurrentAnalyzerRunner.cs [hunk 2]
```diff
@@ -21,38 +20,32 @@ internal class ConcurrentAnalyzerRunner : IAnalyzerRunner
...
-
-            var documents = formattableDocuments.Select(x => x.Document).ToList();
             var result = new CodeAnalysisResult();
-            foreach (var project in solution.Projects)
+            var solution = formattableDocuments.FirstOrDefault().Document.Project.Solution;
+            var documents = formattableDocuments.Select(x => x.Document).ToList();
+            Parallel.ForEach(solution.Projects, project =>
             {
-                var compilation = await project.GetCompilationAsync(cancellationToken);
+                var compilation = project.GetCompilationAsync(cancellationToken).GetAwaiter().GetResult();
```
Comment (JoeRobich) [inline src/Analyzers/ConcurrentAnalyzerRunner.cs]: Should we check if the project is a C# or VB project?

File: src/Analyzers/CodeAnalysisResult.cs (hunk 1 above)
Comment (jmarolf) [reply src/Analyzers/CodeAnalysisResult.cs]: yep.  I need to fix integration tests first.  You can assume that this is the shape of the object you will be given

**Pull request merged**